### PR TITLE
Transfromer and Execution Context Cleanup

### PR DESCRIPTION
**Pull request by indradeep**

### CODE REVIEW

File: blurr/core/session_data_group.py [hunk 1]
```diff
@@ -1,49 +1,84 @@
...
+        super(SessionDataGroup, self).__init__(schema, evaluation_context)
 
     def evaluate(self) -> None:
+        """
+        Overrides the default execution behavior to handle session splits
+        """
+        # Check if current session is stale for the event being processed
         if self.start_time is not None and self.end_time is not None:
-            if not self.schema.split or self.evaluate_expr(self.schema.split):
+            if not self.schema.split or self.schema.split.evaluate(self.global_context, self.local_context):
```
Comment (prashastk) [inline blurr/core/session_data_group.py]: global and local context are no longer available separately.

File: blurr/core/simple_field.py [hunk 2]
```diff
@@ -82,3 +82,10 @@ def type_object(self) -> Any:
     @property
     def default(self) -> Any:
         return set()
+
+
+class SimpleField(Field):
```
Comment (prashastk) [inline blurr/core/simple_field.py]: Why is this needed?

File: blurr/core/streaming_transformer.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+class StreamingTransformer(Transformer):
+    def __init__(self, schema: TransformerSchema, identity, evaluation_context: EvaluationContext) -> None:
+        super().__init__(schema, evaluation_context)
+        self._identity = identity
+        self.evaluation_context.global_add('identity', self._identity)
+
+    def set_source_context(self, source_context: EvaluationContext) -> None:
+        self.evaluation_context.global_context.merge(source_context.global_context)
+        self.evaluation_context.local_include(source_context.local_context)
```
Comment (prashastk) [inline blurr/core/streaming_transformer.py]: Line 46  & 47 are doing the same thing but are using two different approaches. Any reason?

Comment (prashastk) [reply blurr/core/streaming_transformer.py]: Maybe define a function to merge two EvaluationContext?

File: blurr/core/transformer.py [hunk 4]
```diff
@@ -1,72 +1,47 @@
...
+    All transformers inherit from this base.  Adds the current transformer
+    to the context
+    """
 
-    @property
-    def groups(self):
-        return self._groups
+    def __init__(self, schema: TransformerSchema,  evaluation_context: EvaluationContext) -> None:
+        super().__init__(schema, evaluation_context)
+        self.evaluation_context.global_add(self.name, self)
```
Comment (prashastk) [inline blurr/core/transformer.py]: Why is this needed?

File: blurr/runner/local_runner.py [hunk 5]
```diff
@@ -11,17 +11,17 @@ def __init__(self, local_json_files, output_file, stream_dtc_file):
...
 
     def _consume_record(self, record):
-        source_context = Context({'source': record})
+        source_context = EvaluationContext(Context({'source': record}))
         identity = self._transformer_schema.get_identity(source_context)
         if identity not in self._user_transformer:
-            self._user_transformer[identity] = Transformer(
-                self._transformer_schema, identity, self._exec_context)
+            self._user_transformer[identity] = StreamingTransformer(
+                self._transformer_schema, identity, EvaluationContext(self._exec_context, source_context.global_context))
```
Comment (prashastk) [inline blurr/runner/local_runner.py]: Why is source_context a local context?

File: blurr/core/session_data_group.py (hunk 1 above)
Comment (indradeep) [reply blurr/core/session_data_group.py]: Correct. I'll fix this using merge.

File: blurr/core/simple_field.py (hunk 2 above)
Comment (indradeep) [reply blurr/core/simple_field.py]: To maintain consistency of an abstract base, this conforms to the abstract Field/FieldSchema, DataGroup/DataGroupSchema and Transformer/TransformerSchema.  It also sets a good precedent to what the design pattern is.

File: blurr/core/streaming_transformer.py (hunk 3 above)
Comment (indradeep) [reply blurr/core/streaming_transformer.py]: Good idea.

File: blurr/core/transformer.py (hunk 4 above)
Comment (indradeep) [reply blurr/core/transformer.py]: It was in your code... I suppose this is not needed here.

File: blurr/runner/local_runner.py (hunk 5 above)
Comment (indradeep) [reply blurr/runner/local_runner.py]: The way the propagation happens, it doesn't matter if it local or global.
